options.go

```diff
@@ -142,6 +145,9 @@ type FetchOptions struct {
 	// Tags describe how the tags will be fetched from the remote repository,
 	// by default is TagFollowing.
 	Tags TagMode
+	// Force allows the fetch to update a local branch even when the remote
+	// branch does not descend from it.
+	Force bool
```

Comment:
I know this is from someone else's PR, but I don't think this is strictly needed because the `RefSpecs` above already have an `IsForce()` method that says whether that particular ref should be forced or not.  (For `PullOptions`, the default `RefSpec` is defined in the `RemoteConfig`.)  Could you please update it to use the `IsForce` method instead?

Reply:
Will do.

Reply:
Now when actually doing this, are you sure we want this? CheckoutOptions and PullOptions have the Force bool in addition to FetchOptions. Removing it from one but not the others would feel inconsistent.

Reply:
You can keep the `Force` bool if you want, but in that case please also add support for the `IsForce()` method.  The git remote protocol supports both force and non-force fetches in the same batch, so we'll need to be able to specify the force on individual refs.